lib/web_ui/lib/src/engine/skwasm/skwasm_impl/raw/text/raw_paragraph.dart

```diff
@@ -119,3 +119,21 @@ external TextBoxListHandle paragraphGetBoxesForRange(
 @Native<TextBoxListHandle Function(ParagraphHandle)>(
   symbol: 'paragraph_getBoxesForPlaceholders', isLeaf: true)
 external TextBoxListHandle paragraphGetBoxesForPlaceholders(ParagraphHandle handle);
+
+// Returns a list of the code points that were unable to be rendered with the
+// selected fonts. The list is deduplicated, so each code point in the output
+// is unique.
+// If `nullptr` is passed in for `outCodePoints`, we simply return the count
+// of the code points.
+// Note: This must be called after the paragraph has been laid out at least
+// once in order to get valid data.
+@Native<Int Function(
+  ParagraphHandle,
+  Pointer<Uint32>,
+  Int,
+)>(symbol: 'paragraph_getUnresolvedCodePoints', isLeaf: true)
+external int paragraphGetUnresolvedCodePoints(
+  ParagraphHandle handle,
+  Pointer<Uint32> outCodePoints,
+  int outLength,
```

Comment:
`outLength` is the length of the `outCodePoints` buffer that we are passing to the API. This is so that the C++ code doesn't overwrite the buffer into some other region of the memory. However... because we actually got that value from the C++ code in the first place, it is true that we can just omit this parameter altogether and the C++ code can always trust that the buffer is exactly the length that is needed. I can remove it if you think it's confusing, I think this is just my standard "C memory safety" guardrails I automatically added.